assets/script/script.js

```diff
@@ -96,3 +88,125 @@ function renderLeagueStats() {
     }
   }
 }
+
+// get QueryString parameters from welcome page and use it to render results
+const urlParams = new URLSearchParams(window.location.search);
+function loadTeamFromWelcomePage() {
+  for (let i = 0; i < 20; i++) {
+    if (parseInt(urlParams.get("id"), 10) === teamData.teams[i].id) {
+      searchInput.val(teamData.teams[i].name);
+    }
+  }
+  renderTeam();
+  renderLeagueStats();
+  fetchFixtures();
+  searchInput.val("");
+}
+
+// fetch upcoming fixtures
+function fetchFixtures() {
+  fetch(
+    `https://api.football-data.org/v2/teams/${selectedTeamId}/matches?status=SCHEDULED`,
+    {
+      headers: { "X-Auth-Token": "b7bac95afb44489a83d8eb77fc894151" },
+    }
+  )
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (data) {
+      fixturesData = data;
+    })
+    .then(renderFixtures);
+}
+
+// render fixtures
+function renderFixtures() {
+  $("#fixturesContainer").empty();
+  for (i = 0; i < 10; i++) {
+    const fixture = fixturesData.matches[i];
+    if (fixture.competition.name !== "Premier League") {
+      continue;
+    }
+    $("#fixturesContainer").append(
+      $(
+        `<div class="gameContainer">
+          <div class="gameInfo">
+            <ul>
+              <li>${moment
+                .utc(fixture.utcDate)
+                .format("MMMM Do YYYY - HH:mm")}</li>
+              <li><h3>${fixture.competition.name}</h3></li>
+              <li>
+                <span> 
+                  ${fixture.homeTeam.name}
+                </span>
+              </li>
+              <li>
+                <img class='fixtureCrest' src=${homeTeamCrest(fixture)}>
+                <span> 
+                 VS 
+                </span>
+                <img class='fixtureCrest' src=${awayTeamCrest(fixture)}>
+              </li>
+              <li>
+                <span> 
+                ${fixture.awayTeam.name}
+                </span>
+              </li>
+              <li>${stadiumChooser(fixture)}</li>
+            </ul>
+          </div>
+          <div class="map">
+            map
+          </div>
+        </div>`
+      )
+    );
+  }
+}
+
+// function for identifying fixture venue
+function stadiumChooser(fixture) {
+  let homeTeam = fixture.homeTeam.id;
+  for (let team of teamData.teams) {
+    if (team.id === homeTeam) {
+      return team.venue;
+    }
+  }
+}
+
+function homeTeamCrest(fixture) {
+  let homeTeam = fixture.homeTeam.id;
+  for (let team of teamData.teams) {
+    if (team.id === homeTeam) {
+      return team.crestUrl;
+    }
+  }
+}
+
+function awayTeamCrest(fixture) {
+  let awayTeam = fixture.awayTeam.id;
+  for (let team of teamData.teams) {
+    if (team.id === awayTeam) {
+      return team.crestUrl;
+    }
+  }
+}
+
+// toggle fixture display
+const fixtureToggleButton = $("#showFixtures");
+const fixturesContainer = $("#fixturesContainer");
+fixtureToggleButton.click(toggleFixturesDisplay);
+
+function toggleFixturesDisplay() {
```

Comment:
Generally, toggles aren't great because they're not idempotent (meaning if people retry, you'll get different results depending on how many requests successfully get through).

Not that bothered, though.

Reply:
You could have separate buttons for Hide and Show, and change whether they're displayed or not in CSS: that way, you could potentially hit the Hide button a million times, and it wouldn't matter.